### PR TITLE
Storybook: connect-with-site component

### PR DESCRIPTION
**Pull request by Douglashdaniel**
Resolves <https://github.com/brave/brave-browser/issues/14846>

Created connect-with-site component and added to storybook

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. `npm run storybook`
2. Confirm that the storybook shows new component.
3. Check functionality of the new component.

Screen Shot 2021-03-15 at 4 36 32 PM

### CODE REVIEW

**Review by petemill — CHANGES_REQUESTED** [review]
Welcome to the team 🎉 . This looks good, and I really appreciate the storybook-driven development. Only a couple of points of feedback:

- Please use the right element type for interactive elements (e.g. `<button>` for a button) so that we gain all the interactive features of those elements. Where not possible to use the right element type, we can include simple aria labels.
- Please use palette colors so we can more easily identify them. These are all accessible on your elements at `p.theme.palette.x` (e.g. `.blurple300`). They should be identifyable from figma as the designer will also be using tokens from their design system.

And some optional feedback at this point
- I don't see any light / dark mode changes. If this is in the design (hopefully it is but if not maybe work with the designer on it) then there are a couple ways to account for it:
  - a) use `p.theme.color.y` variables which will be different for our light and dark themes
  - b) Make a superset of `IBraveTheme` (e.g. (`IBraveWalletTheme`) which will allow you to make your own named colors which have different light and dark values.
  - c) Simply use `@media (prefers-color-scheme: dark) { ... ` to make changes for light / dark mode.
 - We have some shared components available to use in the brave-ui repo / dependency. However, they were made before the design team finished their design system. We're now in a process to make some of these shared components  up to date with their new designs. Button and Toggle would be safe bets, or anything that we've already used extensively in brave-core as it would all get migrated in one go.

File: components/brave_wallet_ui/components/extension/buttons/nav-button/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import * as React from 'react'
+
+// Styled Components
+import { StyledWrapper, ButtonText } from './style'
+
+export interface Props {
+	theme: 'primary' | 'secondary'
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/buttons/nav-button/index.tsx]: Changed the prop name to "buttonType" I will definitely use the brave-ui button when i can in the future, looking forward to the new spec's!

File: components/brave_wallet_ui/components/extension/buttons/nav-button/style.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+export const StyledWrapper = styled.div<StyleProps>`
+	display: flex;
+	align-items: center;
+	justify-content: center;
+	cursor: pointer;
+	border-radius: 40px;
+	padding: 10px 22px;
+	background-color: ${(p) =>
+		p.buttonTheme === 'primary' ? '#4c54d2' : 'none'};
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/buttons/nav-button/style.ts]: Changed to use theme.pallete, I also added this to some other styles in this commit to keep it consistent!

File: components/brave_wallet_ui/components/extension/buttons/nav-button/style.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import styled from 'styled-components'
+
+interface StyleProps {
+	buttonTheme: 'primary' | 'secondary'
+}
+
+export const StyledWrapper = styled.div<StyleProps>`
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/buttons/nav-button/style.ts]: Changed to use <button> element. In the future I will change this to use the brave-ui shared button!

File: components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts [hunk 4]
```diff
@@ -272,11 +272,13 @@ export default function shieldsPanelReducer (
               }
             )
             chrome.tabs.executeScript(tabData.id, {
-              code: 'try { window.sessionStorage.clear(); } catch(e) {}'})
+              code: 'try { window.sessionStorage.clear(); } catch(e) {}'
```
Comment (petemill) [inline components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts:275]: did you mean to include changes in this file in this PR? If so, please split to a different commit.

File: components/brave_wallet_ui/components/extension/buttons/index.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+import NavButton from './nav-button'
```
Comment (petemill) [inline components/brave_wallet_ui/components/extension/buttons/index.ts:1]: Just a question - what's the reason behind the `/extension` part of the path for these? Seems like at least some of these components would apply to all parts of the wallet UI, so would simply go under `components/brave_wallet_ui/components/...`? And also are we actually using an extension at all or just using bundled WebUI?

File: components/brave_wallet_ui/stories/wallet-extension-panels.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    },
+    {
+      id: '3',
+      name: 'Account 3',
+      address: '0x3f29A1da97149722eB09c526E4eAd698895b426'
+    }
+  ]
+  const selectAccount = (account: WalletAccountType) => {
+    const newList = [...selectedAccounts, account]
+    console.log(newList)
```
Comment (petemill) [inline components/brave_wallet_ui/stories/wallet-extension-panels.tsx:43]: Perhaps either remove thse before merging, or add a string parameter as a label to each. Also if you want to keep it, consider using `console.verbose`?

**Review by petemill — APPROVED** [review]
Thanks for addressing. Would be great to further address some comments before merging but nothing is glaring 😄 

Since it looks like this will be iterative, I've ignored some things that we would want before production, such as consistent dark / light mode changes (the `theme.colors.xyz` will change but the rest won't), more customization of states like focus, etc.

File: components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts:275]: This file had a trailing comma that tslint was complaining about. After i removed and saved prettier formatted these 2 lines.

File: components/brave_wallet_ui/stories/wallet-extension-panels.tsx (hunk 6 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/stories/wallet-extension-panels.tsx]: Removed! :)

File: components/brave_wallet_ui/components/extension/buttons/index.ts (hunk 5 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/buttons/index.ts:1]: Currently I am building out these "Wallet Extension Panels" from Figma. I agree that some of these components will probably need to be reused throughout the wallet UI. Since the wallet ui is still being designed and not yet finalized, its unapparent at the moment what will be reused. But I will definitely address these components paths as more is built out! :) 

Screen Shot 2021-03-19 at 9 09 07 AM

Comment (petemill) [reply components/brave_wallet_ui/components/extension/buttons/index.ts:1]: Oh ok. Maybe we should call the directory "popups" then since it looks like that's what the designer means when they say "extension". It would perhaps be more intuitive for a dev to find where the popup code is since an extension could be a page or a popup etc and maybe we wouldn't even use an extension for the implementation, but a WebUI popup...

File: components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts (hunk 4 above)
Comment (petemill) [reply components/brave_extension/extension/brave_extension/background/reducers/shieldsPanelReducer.ts:275]: Yeah ok in general let's try not to combine unrelated changes to files, especially in otherwise non-related files, in commits. Among the disadvantages are that it makes uplifts harder and file commit history confusing.
